Modules/mathmodule.c

```diff
@@ -2496,6 +2496,92 @@ math_isclose_impl(PyObject *module, double a, double b, double rel_tol,
 }
 
 
+/*[clinic input]
+math.comb
+
+    a: object
+
+    b: object
+
+
+Return the binomial coefficient indexed by the pair of integers n >= k >= 0.
+
+It is the coefficient of kth term in polynomial expansion of the expression
+(1 + x)^n. It is also termed as the number of ways to choose an unordered
+subset of k elements from a fixed set of n elements, usually called
+*n choose k*.
+
+Raises a TypeError if argument(s) are non-integer and ValueError
+if argument(s) are negative or k > n.
+
+[clinic start generated code]*/
+
+static PyObject *
+math_comb_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=1343ae354f5dd2fa input=f5a3952a720899cf]*/
+{
+    if (!(PyLong_Check(a) && PyLong_Check(b))) {
+        PyErr_SetString(PyExc_TypeError,
+            "comb() only accepts integer arguments");
+        return NULL;
+    }
+    a = PyNumber_Long(a);
+    if (a == NULL)
+        return NULL;
+    b = PyNumber_Long(b);
+    if (b == NULL) {
+        Py_DECREF(a);
+        return NULL;
+    }
+
+    PyObject *val = PyLong_FromLong(1),
+        *temp_obj1, *temp_obj2,
+        *dump_var;
+    int overflow;
+    long long i, terms = PyLong_AsLongLongAndOverflow(b, &overflow);
+
+    if (overflow == 1) {
+        PyErr_Format(PyExc_OverflowError,
+                     "minimum(n - k, n) should not exceed %lld",
+                     LLONG_MAX);
+        return NULL;
+    }
+    dump_var = PyLong_FromLong(0);
+    if (PyObject_RichCompareBool(b, (PyObject *)dump_var, Py_LT)) {
+        PyErr_Format(PyExc_ValueError,
+                     "k must be an integer >= 0");
+        return NULL;
+    }
+    if (PyObject_RichCompareBool(a, b, Py_LT)) {
```

Comment:
In the case of an error, this should not overwrite the exception.

Reply:
checking for -1 now, thanks